Module4.3_Search_and_Sorting_Algorithms/activities/Act1_SearchingAndSorting/8.md

```diff
@@ -0,0 +1,37 @@
+<!--title={Matchsticks to Square}-->
+
+<!--badges={10}-->
+
+Now that we have learned how to implement various searching algorithms, use the knowledge and code we have developed throughout this activity to figure out how to solve this problem:
+
+Remember the story of Little Match Girl? Let's imagine that we buy a pack of matchsticks from the girl and, with our matchsticks of assorted lengths, try to find out whether or not we can make one square by using up all the given matchsticks. We cannot *not* break any stick, but we can link them up, and each matchstick must be used *exactly* one time.
+
+Our input will be the given matchsticks we purchased, represented by integers of their lengths. Our output will either be true or false, to represent whether or not we could make one square using all the matchsticks from the pack we bought from the Little Match Girl.
+
+**Note these restrictions:**
+
+- The number of given matchsticks in the inputted array will not exceed `15`.
+- The sum of the lengths of the given matchsticks is in the range of `0` to `10^9`.
+
+![Photography of Piled Red Matchsticks](https://images.pexels.com/photos/1243550/pexels-photo-1243550.jpeg?auto=compress&cs=tinysrgb&h=750&w=1260)
```

Comment:
can you create a custom graphic instead? Like an example of the problem (similar to the one on Leetcode)? I feel like that will provide more clarity for students. I don't think that the written explanation is entirely sufficient on its own.